R/timeseriessummary-data.R

```diff
@@ -524,6 +525,178 @@ parseTSSGapTolerances <- function(reportData, timezone){
+        }
+    }
+  }
+  return(formattedParameters)
+}
+
+#' formats the single point correction parameters
+#' @description formats the single point correction parameters
+#' @param value the value for the parameter single point
+#' @return formatted string of single point parameter for report display
+formatCorrectionsParamSinglePoint <- function(value) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(value)) {
+    formattedParameters <- paste0("SinglePoint ", value)
+  }
+  return(formattedParameters)
+}
+
+#' formats the USGSMultiPoint correction parameters
+#' @description formats the USGSMultiPoint correction parameters
+#' @param startShiftPoints startShift points values and offset as a list
+#' @param endShiftPoints endShift points values and offset as a list
+#' @param usgsType the USGSMultiPoint type
+#' @return formatted string of USGSMultiPoint parameters for report display
+formatCorrectionsParamUSGSMultiPoint <- function(startShiftPoints, endShiftPoints, usgsType) {
+  formattedParameters <- "USGSMultiPoint "
+  if (!isEmptyOrBlank(startShiftPoints) && !isEmptyOrBlank(endShiftPoints) && !isEmptyOrBlank(usgsType)) {
+    startShiftPoints <- as.data.frame(startShiftPoints)
+    endShiftPoints <- as.data.frame(endShiftPoints)
+    if (any(names(startShiftPoints) %in% c("Value","Offset"))) {
+      for (i in 1:nrow(startShiftPoints)) {
+        formattedParameters <- paste0(formattedParameters, "Start shift points value ", startShiftPoints[['Value']][[i]], ", offset ", round(as.numeric(startShiftPoints[['Offset']][[i]],3)), ". ")
+      }
+    }
+    if (any(names(endShiftPoints) %in% c("Value","Offset"))) {
+      for (i in 1:nrow(endShiftPoints)) {
+        formattedParameters <- paste0(formattedParameters, "End shift points value ", endShiftPoints[['Value']][[i]], ", offset ", round(as.numeric(endShiftPoints[['Offset']][[i]],3)), ". ")
+      }
+    }
+    formattedParameters <- paste0(formattedParameters, usgsType)
+  }
+  return(formattedParameters)
+}
+
+#' formats the adjustable trim correction parameters
+#' @description formats the adjustable trim correction parameters
+#' @param upperThresholdPoints upper threshold points date/times and values as a list
+#' @param lowerThresholdPoints lower threshold points date/times and values as a list
+#' @param timezone the timezone for the drift parameters
+#' #' @return formatted string of adjustable trim parameters for report display
+formatCorrectionsParamAdjustableTrim <- function(upperThresholdPoints, lowerThresholdPoints, timezone) {
+  formattedParameters <- "Adjustable trim "
+  if (!isEmptyOrBlank(upperThresholdPoints) || !isEmptyOrBlank(lowerThresholdPoints) && !isEmptyOrBlank(timezone)) {
+    upperThresholdPoints <- as.data.frame(upperThresholdPoints)
+    lowerThresholdPoints <- as.data.frame(lowerThresholdPoints)
+    if (any(names(upperThresholdPoints) %in% c("Value","Time"))) {
```

Comment:
same concern about using `any` here and for the lowerThresholdPoints as in the last function